lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -359,25 +358,20 @@ def write(chunk)
             if record.key?('kubernetes')
               handle_container_metadata(record, entry)
             end
-            # If the log from the user container is json, we want to export it
-            # as a structured log. Now that we've pulled out all the
-            # container-specific metadata from the record, we can replace the
-            # record with the json that the user logged.
-            # To save CPU in the common case of unstructured logs, only check if
-            # the contents are parsable as json for the first entry of each
-            # batch.
-            if is_container_json.nil? && record.key?('log')
-              record_json = parse_json_or_nil(record['log'])
-              if record_json.nil?
-                is_container_json = false
-              else
-                record = record_json
-                is_container_json = true
-              end
-            elsif is_container_json && record.key?('log')
-              record_json = parse_json_or_nil(record['log'])
-              record = record_json unless record_json.nil?
-            end
+          end
+
+          # If the log is json, we want to export it as a structured log unless
+          # there is additional metadata that would be lost.
+          is_json = false
+          if record.length == 1 && record.key?('log')
```

Comment:
My understanding was that in the normal case, additional fields (like "stream") will have been handled and removed already. The purpose of the length check is to prevent clobbering logs that already come in in some kind of structured format, and only handling cases where the remaining data is one line of text. I've noticed that individual text lines from log files sometimes come in a field "log" and sometimes "message" which is why I am using two checks here.
Any suggestions for how I can add a comment to make this clearer?